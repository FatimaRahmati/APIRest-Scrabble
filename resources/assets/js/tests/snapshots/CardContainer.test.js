import { mount } from "@vue/test-utils";
import CardContainer from "@components/CardContainerComponent";

/* Lo normal seria incluirlo en la prueba unitaria (de hecho gran parte del código es el mismo).
   Se separa por razones académicas */
describe("Snapshot: CardContainer", () => {
  const build = (random) => {
    const cards = [];
    const card = {
      order: 0,
      type: "Prueba respuesta server",
      errorCode: 200,
      statusCode: "OK",
      output: "2",
    };

    cards.push(card);

    // fijos
    for (let i = 0; i < 5; i++) {
      let copyCard = Object.assign({}, card); // clono el card
      copyCard.order = i + 1;
      cards.push(copyCard);
    }

    // random
    if (random) {
      const errorCode = [201, 400, 401, 403, 404, 500];
      // el código de error se genera de manera aleatoria
      cards[0].errorCode =
        errorCode[Math.floor(Math.random() * errorCode.length)];
    }

    // hago un mount para cargar tambien los componentes hijos
    const wrapper = mount(CardContainer, {
      propsData: {
        cards: cards,
      },
    });

    return wrapper;
  };

  test("[T013] should render 6 cards", () => {
    const wrapper = build(false);

    expect(wrapper.html()).toMatchSnapshot();
  });

  // objetivamente es el mismo test que el anterior, pero se ha  modificado uno de los cards
  // para que su valor sea aleatorio entre diferentes ejecuciones del test.
  // Eso podria provocar fallos en los snapshots al poder ser diferente el valor cada
  // vez que se ejecuta el test
  // test FAIL
  test("[T014] should render 6 cards (random mode)", () => {
    const wrapper = build(true);

    // deberia comprobar que los datos que se renderizan son siempre los mismos
    // esto puede pasar de manera habitual con elementos que rendericen
    // valores de tiempo relativo, por ejemplo "se conectó por última vez hace XX dias")
    expect(wrapper.html()).toMatchSnapshot();
  });
});
