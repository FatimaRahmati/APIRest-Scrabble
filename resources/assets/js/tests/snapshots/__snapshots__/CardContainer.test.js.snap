// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: CardContainer [T013] should render 5 cards 1`] = `
"<div class=\\"card-container\\">
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      0
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      1
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      2
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      3
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      4
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      5
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>"
`;

exports[`Snapshot: CardContainer [T014] should render 5 cards (random mode) 1`] = `
"<div class=\\"card-container\\">
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      0
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status error\\">
        403 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      1
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      2
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      3
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      4
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      5
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>"
`;

exports[`Snapshot: CardContainer should render 5 cards (random mode) 1`] = `
"<div class=\\"card-container\\">
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      0
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      1
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      2
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      3
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      4
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      5
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>"
`;

exports[`Snapshot: CardContainer should render 5 cards 1`] = `
"<div class=\\"card-container\\">
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      0
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      1
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      2
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      3
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      4
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <div class=\\"output-card\\">
    <div class=\\"info number\\">
      5
    </div>
    <div class=\\"info type\\">
      Prueba respuesta server
    </div>
    <div class=\\"data\\">
      <p>Estado</p>
      <div class=\\"status\\">
        200 OK
      </div>
      <p>Salida</p>
      <div class=\\"output\\">
        <div class=\\"string\\">
          2
        </div>
      </div>
    </div>
  </div>
  <!---->
</div>"
`;
